SimPEG/electromagnetics/time_domain/sources.py

```diff
@@ -878,6 +877,139 @@ def time_nodes(self):
         return self.times
 
 
+class ExponentialWaveform(BaseWaveform):
+    """
+    An exponential ramp-on and linear ramp-off waveform
+
+    Parameters
+    ----------
+    start_time : float, default: -1e-2
+        time at which the transmitter is turned on in units of seconds
+    peak_time : float, default: -1e-3
+        the peak time for the waveform in units of seconds
+    ramp_on_tau : float, default: 1e-3
+        time constant tau in units of seconds controlling how quickly the waveform ramps on (formula: 1-e^(-t/tau))
+
+    Examples
+    --------
+
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+    >>> from SimPEG.electromagnetics import time_domain as tdem
+
+    >>> times = np.linspace(-1e-2, 1e-2, 1000)
+    >>> waveform = tdem.sources.ExponentialWaveform()
+    >>> plt.plot(times, [waveform.eval(t) for t in times])
+    >>> plt.show()
+
+    """
+
+    def __init__(
+        self,
+        start_time=-1e-2,
+        peak_time=-1e-3,
+        ramp_on_tau=1e-3,
+        off_time=0.0,
+        **kwargs,
+    ):
+        super().__init__(hasInitialFields=False, off_time=off_time, **kwargs)
+        self.start_time = start_time
+        self.peak_time = peak_time
+        self.ramp_on_tau = ramp_on_tau
+
+    @property
+    def start_time(self):
+        """Peak time
+
+        Returns
+        -------
+        float
+            The peak time for the Exponential waveform
+        """
+        return self._start_time
+
+    @start_time.setter
+    def start_time(self, value):
+        value = validate_float(
+            "start_time", value, max_val=self.off_time, inclusive_max=False
+        )
+        self._start_time = value
+
+    @property
+    def peak_time(self):
+        """Peak time
+
+        Returns
+        -------
+        float
+            The peak time for the VTEM waveform
+        """
+        return self._peak_time
+
+    @peak_time.setter
+    def peak_time(self, value):
+        value = validate_float(
+            "peak_time",
+            value,
+            min_val=self.start_time,
+            max_val=self.off_time,
+            inclusive_min=False,
+        )
+        self._peak_time = value
+
+    @property
+    def ramp_on_tau(self):
+        """Ramp on rate
+
+        Parameter controlling how quickly the waveform ramps on (default is 3)
```

Comment:
I don't think the default is 3 anymore, best just to remove reference to it here and leave it as described in the constructor.